### ISSUE TITLE
Add open in file explorer 

### ISSUE THREAD

**Issue by mhutti1** [kiwix/kiwix-android#770]
## Feature Request

Allow users a way to open the parent folder of a ZIM file.

**Comment by renggli** [comment]
Turns out this is tricky to implement, as there is no standard way of opening a directory on Android. It can be made to work on certain devices, or if certain third-party apps are installed, but I see no generic way.

This is a similar bug of another android open-source project that describes the wider context of the problem: https://github.com/syncthing/syncthing-android/issues/838

**Comment by mhutti1** [comment]
Oh well, we can keep this open incase a solution is added in the future.

**Comment by kelson42** [comment]
@mhutti1 Do we still really need that? I see no reason (anymore) so far.

**Comment by mhutti1** [comment]
I mean I think it does give them the flexibility to delete ZIM files themselves. Not sure if its technically possible though.

**Comment by kelson42** [comment]
@mhutti1 What I don't understand is "Allow users a way to open the parent folder of a ZIM file."

**Comment by mhutti1** [comment]
As in open the place the ZIM file is located in the file explorer app so they can see its context.

**Comment by kelson42** [comment]
@mhutti1 ZIM file should be deletable from the libary UI... but we don't need to open in a file browser. This is why I strongly suspect this feature request should/will not be implemented. Why making things complex if we can do it easily directly in the library?

**Comment by mhutti1** [comment]
Because I personally think that if we are saving very large files to a users phone we should at least give them an easy way to see exactly what we are doing.

**Comment by kelson42** [comment]
@mhutti1 What kind of information in actually missing in Kiwix's library?

**Comment by mhutti1** [comment]
Where exactly the ZIM files are located.

**Comment by kelson42** [comment]
@mhutti1 We might be able to provide this information without having to implement that feature (just providing the path). I wonder if just saying if the ZIM file is on the "internal" or "external" storage might be enough?

**Comment by mhutti1** [comment]
Not really. Each device can do pretty much what they want, especially on older APIs. Making the path available would probably be ok though.

**Comment by macgills** [comment]
@mhutti1 Is this "For a BookOnDisk list item add an info icon to show extended detail on a book including path"?
We could use a modal bottom sheet and show that when a user clicks on the info icon, show as much information as possible and explicit open/delete(maybe share?) buttons. Would take some UI consideration.

**Comment by mhutti1** [comment]
That sounds like it could work.

**Comment by macgills** [comment]
@kelson42 should I create this `BookDetailSheet`? Bottom sheet examples [here](https://material.io/components/sheets-bottom/#modal-bottom-sheet)

**Comment by kelson42** [comment]
@macgills Yes please.

**Comment by macgills** [comment]
![image](https://user-images.githubusercontent.com/3358282/78151569-a6911680-7430-11ea-98b8-14277f645c73.png)

Here is a very rough implementation with not much formatting of the available date we have on a book

**Comment by macgills** [comment]
@kelson42 here is the current state of this sheet

![image](https://user-images.githubusercontent.com/3358282/80708423-87df7780-8ae3-11ea-9873-29aa96d66704.png)

![image](https://user-images.githubusercontent.com/3358282/80708503-b0677180-8ae3-11ea-8ab4-81af90178fb2.png)

shown on clicking this new `(i)` icon in the library

![image](https://user-images.githubusercontent.com/3358282/80708578-d55be480-8ae3-11ea-99d0-52c661893785.png)

**Comment by macgills** [comment]
@kelson42 I think we should push this from 3.3 I want to make the release sooner rather than later and I think this will need some back and forth, also not a critical feature
